client/document.go

```diff
@@ -119,15 +117,274 @@ func NewDocFromMap(data map[string]any) (*Document, error) {
 	return doc, nil
 }
 
+var jsonArrayPattern = regexp.MustCompile(`^\s*\[.*\]\s*$`)
+
+// IsJSONArray returns true if the given byte array is a JSON Array.
+func IsJSONArray(obj []byte) bool {
+	return jsonArrayPattern.Match(obj)
+}
+
 // NewFromJSON creates a new instance of a Document from a raw JSON object byte array.
-func NewDocFromJSON(obj []byte) (*Document, error) {
-	data := make(map[string]any)
-	err := json.Unmarshal(obj, &data)
+func NewDocFromJSON(obj []byte, sd SchemaDescription) (*Document, error) {
+	doc := newEmptyDoc(sd)
+	err := doc.SetWithJSON(obj)
 	if err != nil {
 		return nil, err
 	}
+	err = doc.generateAndSetDocKey()
+	if err != nil {
+		return nil, err
+	}
+	return doc, nil
+}
+
+// ManyFromJSON creates a new slice of Documents from a raw JSON array byte array.
+func NewDocsFromJSON(obj []byte, sd SchemaDescription) ([]*Document, error) {
+	v, err := fastjson.ParseBytes(obj)
+	if err != nil {
+		return nil, err
+	}
+	a, err := v.Array()
+	if err != nil {
+		return nil, err
+	}
+
+	docs := make([]*Document, len(a))
+	for _, v := range a {
```

Comment:
suggestion:  I would suggest refactoring this block (single doc jsonObj) with `NewDocFromJSON` so that they both use exactly the same logic.  It feels like it would be more maintainable, and it feels strange that they are not doing so.

Reply:
It's not really practical to do so since here we already have a `fastjson.Object` and not a `[]byte`

Reply:
You would probably want to change `NewDocFromJSON` to create a `fastjson.Object` and pass that into the shared code.

Reply:
They already share `setWithFastJSONObject` found within `SetWithJSON`. I don't think it would simplify anything.